### PR TITLE
static graph autogen code support for softmax op

### PR DESCRIPTION
**Pull request by GreatV**
### PR types
Others

### PR changes
Others

### Description

static graph autogen code support for `softmax` op.

### CODE REVIEW

File: paddle/phi/api/yaml/op_compat.yaml [hunk 1]
```diff
@@ -2125,8 +2125,13 @@
...
   inputs :
     x : X
+  outputs :
+    out : Out
+  get_expected_kernel_type :
+    softmax : GetSoftmaxExpectedKernelType
+    softmax_grad : GetSoftmaxGradExpectedKernelType
   extra :
-    attrs : [bool use_mkldnn = false, str mkldnn_data_type = "float32", bool is_test = false]
+    attrs : [str data_format = "AnyLayout", bool use_mkldnn = false, str mkldnn_data_type = "float32", bool is_test = false]
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/op_compat.yaml]: 增加[bool use_cudnn=false]

File: test/cpp/fluid/mkldnn/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,6 @@ cc_test(
   DEPS op_registry
        elementwise_add_op
        activation_op
-       softmax_op
```
Comment (heavyrain-lzy) [inline test/cpp/fluid/mkldnn/CMakeLists.txt:7]: 添加`generated_static_op`

File: test/cpp/fluid/mkldnn/CMakeLists.txt [hunk 3]
```diff
@@ -16,7 +15,6 @@ set(TEST_MKLDNN_CACHING_DEPS
     elementwise_mul_op
     elementwise_add_op
     activation_op
-    softmax_op
```
Comment (heavyrain-lzy) [inline test/cpp/fluid/mkldnn/CMakeLists.txt:19]: 添加`generated_static_op`

File: test/cpp/fluid/CMakeLists.txt [hunk 4]
```diff
@@ -37,7 +37,7 @@ cc_test(
   test_common_infer_shape_functions
   SRCS test_common_infer_shape_functions.cc
   DEPS common_infer_shape_functions ${COMMON_OP_DEPS} activation_op
-       elementwise_add_op softmax_op softmax)
```
Comment (heavyrain-lzy) [inline test/cpp/fluid/CMakeLists.txt:45]: 添加`generated_static_op`

File: test/cpp/fluid/CMakeLists.txt [hunk 5]
```diff
@@ -37,7 +37,7 @@ cc_test(
   test_common_infer_shape_functions
   SRCS test_common_infer_shape_functions.cc
   DEPS common_infer_shape_functions ${COMMON_OP_DEPS} activation_op
```
Comment (heavyrain-lzy) [inline test/cpp/fluid/CMakeLists.txt:44]: `paddle/fluid/operators/unity_build_rule.cmake`里的`softmax_op.cc`删除

**Review by heavyrain-lzy — APPROVED** [review]
LGTM

**Comment by GreatV** [comment]
@luotao1

**Pull request merged**